### PR TITLE
[DO NOT MERGE] Add .NET Core supported env vars and templates.

### PR DESCRIPTION
**Pull request by jerboaa**
- Add supported environment variables.
- Add examples of using the templates.

We should hold off on merging this until the templates are actually there. Perhaps we should split this into two PRs as the environment variables will work as soon as the images get released. The templates section will only work once a release with them being pre-loaded gets out. Ashley, would you be able to take this over?

Thoughts?

cc @omajid @ahardin-rh @tmds

### CODE REVIEW

File: using_images/s2i_images/dot_net_core.adoc [hunk 1]
```diff
@@ -66,8 +66,70 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
+
+.NET Core Environment Variables
+[cols="4a,6a,6a",options="header"]
+|===
+
+|Variable Name |Description |Default
+
+|`*DOTNET_STARTUP_PROJECT*`
+|Used to select the project to run. This must be the folder containing `project.json`.
+|Current working directory, a.k.a `.`
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: Suggestion. Remove 'Current working directory, a.k.a'.

File: using_images/s2i_images/dot_net_core.adoc [hunk 2]
```diff
@@ -66,8 +66,70 @@ link:https://github.com/redhat-developer/s2i-dotnetcore/blob/master/dotnet_image
...
+ For `project.json`, the assembly name defaults to the `project.json` parent folder.
+ When `project.json` is at the `context-dir`, the parent folder name will be `src`. So, by
+ default, this generates a `src.dll` assembly. Setting `DOTNET_ASSEMBLY_NAME` will cause:
+
+  - the assembly to be <DOTNET_ASSEMBLY_NAME>.dll
+  - the application sources to be in subfolder `DOTNET_ASSEMBLY_NAME` in the deployed container.
+
+|The name of the
+ parent folder of the folder with `project.json`.
+
```
Comment (tmds) [inline using_images/s2i_images/dot_net_core.adoc]: Suggestion: `The name of the DOTNET_STARTUP_PROJECT folder`

**Comment by tmds** [comment]
:+1:

**Comment by omajid** [comment]
lgtm

**Comment by ahardin-rh** [comment]
@jerboaa Thanks! Happy to help. Yes, I agree that this should be split into two separate PRs if the information needs released at different times. I can take it from here and incorporate some minor edits. What is the targeted release date for each piece?

**Comment by jerboaa** [comment]
@ahardin-rh Thanks! The environment variables section should be appropriate for ose 3.3/3.4 once the new images land in the registry. That'll probably happen today. The tags which have them are `dotnet/dotnetcore-10-rhel7:1.0-14` and `dotnet/dotnetcore-11-rhel7:1.1-5`.

The imagestream/templates part, I don't know. We need to QA it once the images have been released and then OpenShift folks need to get it into a release. I can tell you more once I've handed it over to them to incorporate.

**Comment by ahardin-rh** [comment]
@jerboaa @omajid Thanks! This work is now continued in:
https://github.com/openshift/openshift-docs/pull/3966 (. NET Core Templates)
https://github.com/openshift/openshift-docs/pull/3965 (environment variables)

**Pull request closed without merging**